src/io/dataset_loader.cpp

```diff
@@ -594,7 +594,22 @@ Dataset* DatasetLoader::CostructFromSampleData(double** sample_values,
 
   const data_size_t filter_cnt = static_cast<data_size_t>(
     static_cast<double>(config_.min_data_in_leaf * total_sample_size) / num_data);
-  if (Network::num_machines() == 1) {
+
+  bool force_findbin_in_single_machine = false;
+  if (Network::num_machines() > 1) {
+    int total_num_feature = Network::GlobalSyncUpByMin(num_col);
+    size_t esimate_sync_size = BinMapper::SizeForSpecificBin(config_.max_bin) * total_num_feature;
```

Comment:
This will still lead to overflow, need to cast the operands before assigning to a wider type.

```suggestion
    size_t esimate_sync_size = (size_t)BinMapper::SizeForSpecificBin(config_.max_bin) * (size_t)total_num_feature;
```